### PR TITLE
MGMT-16528: Added openshift template, build script and makefile

### PR DESCRIPTION
**Pull request by omer-vishlitzky**
[MGMT-16528](https://issues.redhat.com//browse/MGMT-16528) - Move jira-unfurler app to app-sre clusters.

Added openshift template, build script and makefile

### CODE REVIEW

File: openshift/template.yaml [hunk 1]
```diff
@@ -0,0 +1,74 @@
+apiVersion: template.openshift.io/v1
+kind: Template
+metadata:
+  name: jira-unfurl-bot
+objects:
+- apiVersion: apps/v1
+  kind: Deployment
+  metadata:
+    name: jira-unfurl-bot-deployment
```
Comment (danmanor) [inline openshift/template.yaml:9]: Lets remove the `-deployment`

**Review by danmanor — CHANGES_REQUESTED** [review]
Did you test it in integration / locally ?

**Comment by danmanor** [comment]
@omer-vishlitzky Please provide a description, reference the Jira issue key in the commit message prefix and PR header prefix

**Comment by omer-vishlitzky** [comment]
I have tested it locally, yes.

File: openshift/template.yaml [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+                  name: ${JIRA_UNFURL_BOT_SECRETS_NAME}
+            - name: JIRA_ACCESS_TOKEN
+              valueFrom:
+                secretKeyRef:
+                  key: ${JIRA_UNFURL_BOT_JIRA_ACCESS_TOKEN}
+                  name: ${JIRA_UNFURL_BOT_SECRETS_NAME}
+
+parameters:
+- name: IMAGE_NAME
+  value: "quay.io/ovishlit/jira-unfurl-bot"
```
[the remark this answers is not in the document]
Comment (omer-vishlitzky) [reply openshift/template.yaml]: done

File: openshift/template.yaml (hunk 1 above)
Comment (omer-vishlitzky) [reply openshift/template.yaml]: done

**Comment by danmanor** [comment]
@omer-vishlitzky We need the commit message prefix as well

File: openshift/template.yaml [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+  value: "2000Mi"
+- name: CPU_REQUEST
+  value: "300m"
+- name: MEMORY_REQUEST
+  value: "400Mi"
+- name: JIRA_UNFURL_BOT_SECRETS_NAME
+  value: "jira-unfurl-bot-secrets"
+- name: JIRA_UNFURL_BOT_SLACK_BOT_TOKEN 
+  value: "slack-bot-token"
+- name: JIRA_UNFURL_BOT_SLACK_APP_TOKEN
```
Comment (danmanor) [inline openshift/template.yaml:65]: Duplication

**Comment by danmanor** [comment]
@omer-vishlitzky Did you test that all the Makefile target actually work ?

File: openshift/template.yaml (hunk 3 above)
Comment (omer-vishlitzky) [reply openshift/template.yaml:65]: bot token vs app token

**Pull request merged**